packages/build-utils/src/fs/read-config-file.ts

```diff
@@ -30,7 +30,9 @@ export async function readConfigFile<T>(
     if (data) {
       const str = data.toString('utf8');
       try {
-        if (name.endsWith('.json')) {
+        if (name === 'yarn.lock') {
+          return yaml.safeLoad(str, { filename: name }) as T;
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (blocking):** ‏add tests

Can you add tests for this behavior?